packages/sdk-router/src/sdk.ts

```diff
@@ -10,170 +10,563 @@ import {
+      CCTP_ROUTER_ADDRESS[destChainId]
+    ) {
+      const originCCTPRouter = this.synapseCCTPRouters[originChainId]
+      const destCCTPRouter = this.synapseCCTPRouters[destChainId]
+      cctpQuotePromise = this.calculateBestQuote(
+        originCCTPRouter,
+        destCCTPRouter,
+        destChainId,
+        tokenIn,
+        tokenOut,
+        amountIn,
+        deadline
+      ).catch((error) => {
+        console.error('Error with cctpQuotePromise', error)
+        return undefined
+      })
+    } else {
+      cctpQuotePromise = Promise.resolve(undefined)
+    }
+
+    const [synapseQuote, cctpQuote] = await Promise.all([
+      synapseQuotePromise,
+      cctpQuotePromise,
+    ])
+
+    const bestQuote = [synapseQuote, cctpQuote].reduce(
+      (prev, current) =>
+        !prev ||
+        (current &&
+          current.maxAmountOut &&
+          prev.maxAmountOut &&
+          current.maxAmountOut.gt(prev.maxAmountOut))
+          ? current
+          : prev,
+      undefined
+    )
+
+    if (!bestQuote) {
+      throw new Error('No route found')
+    }
+
+    return bestQuote
+  }
+
+  /**
+   * This method calculates the best bridge quote by using either the SynapseRouter or SynapseCCTPRouter.
+   * For each set of bridge tokens, it fetches queries from the origin and destination routers and finds the best query.
+   * After that, it finalizes the quote and checks if this quote is better than the previous best.
+   *
+   * @param originRouter - The origin router (SynapseRouter or SynapseCCTPRouter or undefined).
+   * @param destRouter - The destination router (SynapseRouter or SynapseCCTPRouter or undefined).
+   * @param destChainId - The ID of the destination chain.
+   * @param tokenIn - The input token.
+   * @param tokenOut - The output token.
+   * @param amountIn - The amount of input token.
+   * @param deadline - The transaction deadline, optional.
+   *
+   * @returns - A promise that resolves to the best bridge quote, or undefined if no best quote could be determined.
+   */
+  private async calculateBestQuote(
+    originRouter: SynapseRouter | SynapseCCTPRouter | undefined,
+    destRouter: SynapseRouter | SynapseCCTPRouter | undefined,
+    destChainId: number,
+    tokenIn: string,
+    tokenOut: string,
+    amountIn: BigintIsh,
+    deadline?: BigNumber
+  ): Promise<BridgeQuote | undefined> {
+    if (!originRouter || !destRouter) {
+      return
+    }
+
+    let bestQuote: BridgeQuote | undefined
+
+    // Getting bridge tokens from cache or fetch from destination router
+    const bridgeTokensArray = await this.getBridgeTokens(
+      destChainId,
+      tokenOut,
+      destRouter
+    )
+
+    // Iterate through each array of bridge tokens
+    for (const bridgeTokens of bridgeTokensArray) {
+      try {
+        // Fetching queries from origin router
+        const originQueries = await this.getOriginQueries(
+          originRouter,
+          tokenIn,
+          [bridgeTokens.symbol],
```

Comment:
Why do we iterate through array of bridge tokens, instead of using the array of their symbols here? Passing a full array instead of `[bridgeTokens.symbol]` should limit the amount of RPC calls to 1 no matter the amount of possible bridge symbols.